scheduler/load-appointments-on-demand/load-appointments-on-demand/Services/SchedulerLodService.cs

```diff
@@ -0,0 +1,227 @@
+            switch (currentView)
+            {
+                case SchedulerView.Day:
+                    break;
+                case SchedulerView.MultiDay:
+                    endDate = endDate.AddDays(multiDayDaysCount);
+                    break;
+                case SchedulerView.Week:
+                    startDate = GetCurrentWeekStartTime(startDate);
+                    endDate = startDate.AddDays(7);
+                    break;
+                case SchedulerView.Month:
+                    // if adjacent months are visible, it is up to +/- 6 days
+                    // to optimize futher you'd have to write a lot of calendar logic to find
+                    // what day of the week the 1st of the month is, and how many days from the previous
+                    // and next month are visible, which can make for convoluted and complex code
+                    startDate = new DateTime(startDate.Year, startDate.Month, 1).AddDays(-6);
+                    // make it even simpler - no months are shorter than 28 days, none is longer than 31
+                    // so adding 9 adds at least 6 to the longest possible for the case where most days are seen
+                    endDate = new DateTime(startDate.Year, startDate.Month, 28).AddDays(-9);
+                    break;
+                default:
+                    throw new ArgumentException("the service does not know how to handle this scheduler view yet");
+            }
+
+            // get appointments from the database
+            return await GetFilteredAppointmentsPerRangeFromDataBaseAsync(startDate, endDate);
+        }
+
+        private async Task<List<SchedulerAppointment>> GetFilteredAppointmentsPerRangeFromDataBaseAsync(DateTime startDate, DateTime endDate)
+        {
+            // filter appointments by the time required range 
+            // optimize this query as required by your data source
+            // for example, pass arguments to the method, use stored procedures and so on
+            // this here showcases the basic concept of filtering the data on the server, not in the UI
+            // you need to implement it according to your actual optimization techniques and project
+            // in the real app you should filter and optimize here, this is not implemented in this sample
+            List<SchedulerAppointment> appointments = await GetDummyAppointments();
+            // here we just filter in memory for demonstration purposes because it is easy to follow in the example
+            // do NOT do this in the real app - this is the equivalend of selecting all rows, getting them to the app and filtering here
+            appointments = appointments.Where(a =>
+                (a.Start.Date >= startDate.Date && a.Start.Date <= endDate.Date) || //starts in the current range
+                (a.End.Date >= startDate.Date && a.End.Date <= endDate.Date)        //OR ends in the current range
+            ).ToList();

```

Comment:
The correct check is that an appointment Starts before the EndRange AND Ends after StartRange.
```suggestion
            appointments = appointments.Where(a =>
                (a.Start.Date < endData.Date && a.End.Date > startDate.Date)     //OR ends in the current range
            ).ToList();
```